File: internal/model/post.go
package model

import (
	"time"

	"github.com/google/uuid"
)

const (
	PostStatusDraft     = "draft"
	PostStatusPublished = "published"
	PostStatusArchived  = "archived"
)

type Post struct {
	BaseModel
	Title           string     `gorm:"size:255;not null" json:"title"`
	Slug            string     `gorm:"size:255;not null;uniqueIndex" json:"slug"`
	Summary         string     `gorm:"type:text" json:"summary"`
	ContentMarkdown string     `gorm:"type:text;not null" json:"content_markdown"`
	ContentHTML     string     `gorm:"type:text;not null" json:"content_html"`
	Status          string     `gorm:"size:32;not null;index" json:"status"`
	CoverImage      string     `gorm:"size:1024" json:"cover_image"`
	Password        string     `gorm:"size:255" json:"-"`
	PublishedAt     *time.Time `json:"published_at"`
	ShowComments    bool       `gorm:"not null;default:true" json:"show_comments"`
	CategoryID      *uuid.UUID `gorm:"type:char(36);index" json:"category_id"`
	Category        *Category  `json:"category,omitempty"`
	Tags            []Tag      `gorm:"many2many:post_tags;" json:"tags,omitempty"`
}

File: ui/admin/src/types/post.ts
/**
 * 文章相关类型定义
 */

/** 文章状态枚举 */
export type PostStatus = 'published' | 'draft' | 'archived'

/** 文章数据结构（与后端对齐） */
export interface Post {
  id: string
  title: string
  slug: string
  summary: string
  category: { id: string; name: string; slug: string } | null
  categoryId: string | null
  tags: { id: string; name: string; slug: string }[]
  status: PostStatus
  coverImage: string
  showComments: boolean
  hasPassword: boolean
  hasProtected: boolean
  createdAt: string
  updatedAt: string
  publishedAt: string | null
}

/** 文章列表查询参数 */
export interface PostQueryParams {
  page: number
  pageSize: number
  keyword?: string
  status?: PostStatus | 'all'
  categoryId?: string
  tagId?: string
}

/** 分页响应 */
export interface PaginatedData<T> {
  items: T[]
  pagination: {
    page: number
    pageSize: number
    total: number
  }
}

/** 文章详情（含正文内容） */
export interface PostDetail extends Post {
  contentMarkdown: string
  contentHtml: string
}

/** 文章表单数据 */
export interface PostFormData {
  title: string
  slug: string
  summary: string
  contentMarkdown: string
  categoryId: string
  tagIds: string[]
  status: PostStatus
  coverImage: string
  password: string
}
